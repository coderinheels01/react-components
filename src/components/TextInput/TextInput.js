import React from "react";
import PropTypes from "prop-types";
import Label from "../Label";
import styled from "styled-components";

export const TextInputWrapper = styled.div`
  margin-bottom: 16;
`;

export const TextInputInner = styled.input(props => ({
  border: props.error ? "solid 1px red" : ""
}));

export const Error = styled.div`
  color: red;
`;

const TextInput = ({
  htmlId,
  label,
  required = false,
  name,
  type = "text",
  placeholder,
  children,
  onChange,
  value,
  error,
  ...props
}) => (
  <TextInputWrapper>
    <Label htmlFor={htmlId} label={label} required={required} />
    <TextInputInner
      id={htmlId}
      type={type}
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      error={error}
      {...props}
    />
    {children}
    {error && <Error>{error}</Error>}
  </TextInputWrapper>
);

TextInput.propTypes = {
  /** HTML id for tying label and input together. Used for testing purpose*/
  htmlId: PropTypes.string.isRequired,

  /** Input label */
  label: PropTypes.string.isRequired,

  /** Function to call on change */
  onChange: PropTypes.func.isRequired,

  /** Mark input with asterisk if it is required */
  required: PropTypes.bool,

  /** Input name */
  name: PropTypes.string,

  /** Input type */
  type: PropTypes.oneOf(["text", "password", "number"]),

  /** Placeholder to display when empty */
  placeholder: PropTypes.string,

  /** Child components to display next to the input*/
  children: PropTypes.node,

  /** Input value*/
  value: PropTypes.any,

  /** Error string to display when error occured*/
  error: PropTypes.string
};

export default TextInput;
