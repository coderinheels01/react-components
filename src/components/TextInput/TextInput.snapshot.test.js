import React from "react";
import renderer from "react-test-renderer";
import TextInput from "./TextInput";

describe("TextInput", () => {
  let props;
  let textInputTree;

  describe("TextInput with type text", () => {
    beforeEach(() => {
      props = {
        htmlId: "test html id ",
        label: "test label",
        required: true,
        name: "test name",
        type: "text",
        placeholder: "test placeholder",
        children: <div />,
        onChange: jest.fn(),
        value: "test value",
        error: "some error"
      };
      textInputTree = renderer.create(<TextInput {...props} />);
    });

    it("render TextInput of type text", () => {
      expect(textInputTree).toMatchSnapshot();
    });

    afterEach(() => {
      textInputTree.unmount();
    });
  });

  describe("TextInput with type password", () => {
    beforeEach(() => {
      props = {
        htmlId: "test html id ",
        label: "test label",
        required: true,
        name: "test name",
        type: "password",
        placeholder: "test placeholder",
        children: <div />,
        onChange: jest.fn(),
        value: "test value",
        error: "some error"
      };
      textInputTree = renderer.create(<TextInput {...props} />);
    });

    it("render TextInput of type password", () => {
      expect(textInputTree).toMatchSnapshot();
    });
    afterEach(() => {
      textInputTree.unmount();
    });
  });
});
