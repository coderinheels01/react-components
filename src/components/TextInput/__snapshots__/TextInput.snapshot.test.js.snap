// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput TextInput with type password render TextInput of type password 1`] = `
<div
  className="sc-dlfnbm InidV"
>
  <label
    className="sc-bdfBwQ gaSFkq"
    htmlFor="test html id "
  >
    test label
    <span
      className="sc-gsTCUz wIMML"
    >
       * 
    </span>
  </label>
  <input
    className="sc-hKgILt brxkeH"
    id="test html id "
    name="test name"
    onChange={[MockFunction]}
    placeholder="test placeholder"
    type="password"
  />
  <div />
  <div
    className="sc-eCssSg dIgmt"
  >
    some error
  </div>
</div>
`;

exports[`TextInput TextInput with type text render TextInput of type text 1`] = `
<div
  className="sc-dlfnbm InidV"
>
  <label
    className="sc-bdfBwQ gaSFkq"
    htmlFor="test html id "
  >
    test label
    <span
      className="sc-gsTCUz wIMML"
    >
       * 
    </span>
  </label>
  <input
    className="sc-hKgILt brxkeH"
    id="test html id "
    name="test name"
    onChange={[MockFunction]}
    placeholder="test placeholder"
    type="text"
  />
  <div />
  <div
    className="sc-eCssSg dIgmt"
  >
    some error
  </div>
</div>
`;
